pandas/tests/plotting/test_hist_method.py

```diff
@@ -659,13 +659,13 @@ def test_grouped_hist_layout(self, hist_df):
         with pytest.raises(ValueError, match=msg):
             df.hist(column="height", by=df.category, layout=(-1, -1))
 
-        with tm.assert_produces_warning(UserWarning):
+        with tm.assert_produces_warning(UserWarning, check_stacklevel=False):
             axes = _check_plot_works(
                 df.hist, column="height", by=df.gender, layout=(2, 1)
             )
```

Comment:
Separately it'd be good to have `_check_plot_works_check_warnings` (analogous to `parser.read_csv_check_warnings`), but for now I'm just ignoring the stacklevel check